Add route for fetching supported languages per application

gmisail opened the pull request:
This change will allow frontends to fetch a list of languages that are supported by a given application. For instance, an "agent" supports 5 languages while a "complainant" supports 10; in both cases, they can use the `/languages/{app_id}` route to get the languages that are allowed by that application. 

## Description
In this PR, the database has been modified to add support for storing languages along with their corresponding supported applications. JPA entities have been added to reflect these changes, as well as routes to fetch the data. Tests were also added to the routes, mapper, and service.

<!-- insert jira markdown link with title -->
[(API) Add route to retrieve supported languages by application](https://dol-ewf.atlassian.net/browse/WP-180)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
This change is required in order to display which languages are supported on the frontend. For instance, when you register for an account you will need to choose a preferred langauge; this feature is necessary to know which languages to display.

## Areas for Focus
<!--- if applicable, call out areas where you think a reviewer should give extra focus --->
<!--- perhaps an edge case scenario you're not sure how to handle, a design decision you made because you weren't sure what the right decision was, a test that you notice fails sometimes, etc --->

## How Has This Been Tested?
<!--- Describe how you tested your changes (including automated tests run and manual testing done) in enough detail that a reviewer can reproduce. -->
The changes have unit tests that check the functionality of the API route, mapper, and language service. 

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [x] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If any do not apply, leave a description indicating why -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
<!-- - [ ] My change requires a version increment in gradle.properties -->
<!-- - [ ] My change requires redeployment of the API Gateway Deployment after the pipeline succeeds -->
- [x] I have confirmed all acceptance criteria defined by the story have been met

JayStGelais commented on service/src/main/java/io/nuvalence/ds4g/user/api/service/controller/LanguageApiDelegateImpl.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public Optional<NativeWebRequest> getRequest() {
+        return LanguagesApiDelegate.super.getRequest();
```
Is it necessary necessary to override this method with thisimplementation? Isn't this functionally the same as not overriding it?

gmisail replied:
You're right, it is unnecessary. Removed.

gmisail commented on service/src/main/java/io/nuvalence/ds4g/user/api/service/service/ApplicationLanguageService.java:
```diff
@@ -0,0 +1,52 @@
+     * @param id Application ID.
+     * @return List of languages supported by the application.
+     */
+    public ResponseEntity<List<LanguageDTO>> getSupportedLanguagesById(UUID id) {
```
I changed this so that it will return a list of LanguageEntity's, as opposed to a ResponseEntity of a list of DTO's. This makes it more suitable for use cases aside from REST API's, as you have mentioned.

gmisail commented:
I fixed those minor code style issues and removed the unnecessary code. I also modified the relationship between the service and controller, moving the mapping layer from one to the other. Now that they're no longer coupled, the service methods will be more suitable for use elsewhere in the code. To reflect these changes, I also updated the tests for the API and service.
